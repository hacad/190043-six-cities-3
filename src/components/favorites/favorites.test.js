import React from "react";
import renderer from "react-test-renderer";
import Favorites from "./favorites.jsx";

jest.mock(`../../hocs/with-authorization/with-authorization.js`, () => () => `<div />`);

it(`Favorites correctly renders after relaunch`, () => {
  // Arrange && Act
  const favorites = renderer.create(<Favorites />).toJSON();

  // Assert
  expect(favorites).toMatchSnapshot();
});
