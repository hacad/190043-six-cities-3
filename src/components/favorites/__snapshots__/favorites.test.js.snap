// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorites empty correctly renders after relaunch 1`] = `
Array [
  <div
    style={
      Object {
        "display": "none",
      }
    }
  >
    <svg
      xmlns="http://www.w3.org/2000/svg"
    >
      <symbol
        id="icon-arrow-select"
        viewBox="0 0 7 4"
      >
        <path
          clipRule="evenodd"
          d="M0 0l3.5 2.813L7 0v1.084L3.5 4 0 1.084V0z"
          fillRule="evenodd"
        />
      </symbol>
      <symbol
        id="icon-bookmark"
        viewBox="0 0 17 18"
      >
        <path
          d="M3.993 2.185l.017-.092V2c0-.554.449-1 .99-1h10c.522 0 .957.41.997.923l-2.736 14.59-4.814-2.407-.39-.195-.408.153L1.31 16.44 3.993 2.185z"
        />
      </symbol>
      <symbol
        id="icon-star"
        viewBox="0 0 13 12"
      >
        <path
          clipRule="evenodd"
          d="M6.5 9.644L10.517 12 9.451 7.56 13 4.573l-4.674-.386L6.5 0 4.673 4.187 0 4.573 3.549 7.56 2.483 12 6.5 9.644z"
          fillRule="evenodd"
        />
      </symbol>
    </svg>
  </div>,
  <<div /> />,
  <div
    className="page"
  >
    <main
      className="page__main page__main--favorites page__main--favorites-empty"
    >
      <div
        className="page__favorites-container container"
      >
        <section
          className="favorites favorites--empty"
        >
          <h1
            className="visually-hidden"
          >
            Favorites (empty)
          </h1>
          <div
            className="favorites__status-wrapper"
          >
            <b
              className="favorites__status"
            >
              Nothing yet saved.
            </b>
            <p
              className="favorites__status-description"
            >
              Save properties to narrow down search or plan yor future trips.
            </p>
          </div>
        </section>
      </div>
    </main>
    <footer
      className="footer container"
    >
      <a
        className="footer__logo-link"
        href="main.html"
      >
        <img
          alt="6 cities logo"
          className="footer__logo"
          height="33"
          src="img/logo.svg"
          width="64"
        />
      </a>
    </footer>
  </div>,
]
`;

exports[`Favorites not empty correctly renders after relaunch 1`] = `
Array [
  <div
    style={
      Object {
        "display": "none",
      }
    }
  >
    <svg
      xmlns="http://www.w3.org/2000/svg"
    >
      <symbol
        id="icon-arrow-select"
        viewBox="0 0 7 4"
      >
        <path
          clipRule="evenodd"
          d="M0 0l3.5 2.813L7 0v1.084L3.5 4 0 1.084V0z"
          fillRule="evenodd"
        />
      </symbol>
      <symbol
        id="icon-bookmark"
        viewBox="0 0 17 18"
      >
        <path
          d="M3.993 2.185l.017-.092V2c0-.554.449-1 .99-1h10c.522 0 .957.41.997.923l-2.736 14.59-4.814-2.407-.39-.195-.408.153L1.31 16.44 3.993 2.185z"
        />
      </symbol>
      <symbol
        id="icon-star"
        viewBox="0 0 13 12"
      >
        <path
          clipRule="evenodd"
          d="M6.5 9.644L10.517 12 9.451 7.56 13 4.573l-4.674-.386L6.5 0 4.673 4.187 0 4.573 3.549 7.56 2.483 12 6.5 9.644z"
          fillRule="evenodd"
        />
      </symbol>
    </svg>
  </div>,
  <<div /> />,
  <div
    className="page"
  >
    <main
      className="page__main page__main--favorites"
    >
      <div
        className="page__favorites-container container"
      >
        <section
          className="favorites"
        >
          <h1
            className="favorites__title"
          >
            Saved listing
          </h1>
          <ul
            className="favorites__list"
          >
            <li
              className="favorites__locations-items"
            >
              <div
                className="favorites__locations locations locations--current"
              >
                <div
                  className="locations__item"
                >
                  <a
                    className="locations__item-link"
                    href="#"
                  >
                    <span>
                      Amsterdam
                    </span>
                  </a>
                </div>
              </div>
              <div
                className="favorites__places"
              >
                <article
                  className="favorites__card place-card"
                >
                  <div
                    className="favorites__image-wrapper place-card__image-wrapper"
                  >
                    <a
                      href="#"
                    >
                      <img
                        alt="Place image"
                        className="place-card__image"
                        height="110"
                        src="img/apartment-01.jpg"
                        width="150"
                      />
                    </a>
                  </div>
                  <div
                    className="favorites__card-info place-card__info"
                  >
                    <div
                      className="place-card__price-wrapper"
                    >
                      <div
                        className="place-card__price"
                      >
                        <b
                          className="place-card__price-value"
                        >
                          €
                          120
                        </b>
                        <span
                          className="place-card__price-text"
                        >
                          / night
                        </span>
                      </div>
                      <button
                        className="place-card__bookmark-button place-card__bookmark-button--active button"
                        type="button"
                      >
                        <svg
                          className="place-card__bookmark-icon"
                          height="19"
                          width="18"
                        >
                          <use
                            xlinkHref="#icon-bookmark"
                          />
                        </svg>
                        <span
                          className="visually-hidden"
                        >
                          In bookmarks
                        </span>
                      </button>
                    </div>
                    <div
                      className="place-card__rating rating"
                    >
                      <div
                        className="place-card__stars rating__stars"
                      >
                        <span
                          style={
                            Object {
                              "width": "93%",
                            }
                          }
                        />
                        <span
                          className="visually-hidden"
                        >
                          Rating
                        </span>
                      </div>
                    </div>
                    <h2
                      className="place-card__name"
                    >
                      <a
                        href="/offer/1"
                        onClick={[Function]}
                      >
                        Beautiful &amp; luxurious apartment at great location
                      </a>
                    </h2>
                    <p
                      className="place-card__type"
                    >
                      apartment
                    </p>
                  </div>
                </article>
              </div>
            </li>
          </ul>
        </section>
      </div>
    </main>
    <footer
      className="footer container"
    >
      <a
        className="footer__logo-link"
        href="main.html"
      >
        <img
          alt="6 cities logo"
          className="footer__logo"
          height="33"
          src="img/logo.svg"
          width="64"
        />
      </a>
    </footer>
  </div>,
]
`;
